modules/custom-status/custom-status.php

```diff
@@ -472,47 +524,67 @@ public function remove_or_add_publish_capability_for_user( $allcaps, $cap, $args
 	 *
 	 * @return object|WP_Error $inserted_term The newly inserted term object or a WP_Error object
 	 */
-	public function add_custom_status( $term, $args = [] ) {
-		// Term is always added to the end of the list
-		$default_position = count( $this->get_custom_statuses() ) + 1;
+	public function add_custom_status( array $args ): WP_Term|WP_Error {
+		if ( ! isset( $args['position'] ) ) {
+			// get the existing statuses, ordered by position
+			$custom_statuses = $this->get_custom_statuses();
 
-		$slug = ( ! empty( $args['slug'] ) ) ? $args['slug'] : sanitize_title( $term );
-		unset( $args['slug'] );
+			// get the last status position
+			$last_position = $custom_statuses[ array_key_last( $custom_statuses ) ]->meta[ self::METADATA_POSITION_KEY ];
 
-		if ( ! isset( $args['position'] ) ) {
-			$args['position'] = $default_position;
+			// set the new status position to be one more than the last status
+			$args['position'] = $last_position + 1;
 		}
 
-		$required_user_ids = [];
-		if ( isset( $args['required_user_ids'] ) ) {
-			$required_user_ids = $args['required_user_ids'];
-			unset( $args['required_user_ids'] );
-		}
+		$term_to_save = [
+			'slug'        => $args['slug'] ?? sanitize_title( $args['name'] ),
+			'description' => $args['description'] ?? '',
+		];
 
-		$encoded_description = TaxonomyUtilities::get_encoded_description( $args );
+		$term_name = $args['name'];
 
-		$inserted_term = wp_insert_term( $term, self::TAXONOMY_KEY, [
-			'slug'        => $slug,
-			'description' => $encoded_description,
-		] );
+		$inserted_term = wp_insert_term( $term_name, self::TAXONOMY_KEY, $term_to_save );
+
+		if ( is_wp_error( $inserted_term ) ) {
+			return $inserted_term;
+		}
 
 		// Reset our internal object cache
 		$this->custom_statuses_cache = [];
 
-		if ( is_wp_error( $inserted_term ) ) {
-			return $inserted_term;
+		$term_id = $inserted_term['term_id'];
+
+		$position = $args['position'];
+		$required_metadata_fields = $args['required_metadata_fields'] ?? [];
+		$required_user_ids = $args['required_user_ids'] ?? [];
+
+		// In case of failure, data cleanup happens which includes the term and the meta keys.
```

Comment:
This ensures we don't have bad data sitting around that we need to handle.

Ideally, once all our features are clean we should have a regular cron job that goes in and deletes all the bad data that could exist.